### PR TITLE
#2621 Add a dynawo test case for IEC WT4A

### PR DESCRIPTION
**Pull request by GerardThevenon**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

File: examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.dyd [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    This file is part of Dynawo, an hybrid C++/Modelica open source suite of
+    simulation tools for power systems.
+-->
+<dyn:dynamicModelsArchitecture xmlns:dyn="http://www.rte-france.com/dynawo">
+  <dyn:blackBoxModel id="WT4A" lib="IECWT4ACurrentSource" parFile="WT4ACurrentSource.par" parId="WT4A"/>
+  <dyn:blackBoxModel id="PRefPu" lib="Step" parFile="WT4ACurrentSource.par" parId="PRefPu"/>
+  <dyn:blackBoxModel id="QRefPu" lib="Step" parFile="WT4ACurrentSource.par" parId="QRefPu"/>
+  <dyn:blackBoxModel id="TanPhi" lib="Step" parFile="WT4ACurrentSource.par" parId="TanPhi"/>
+  <dyn:blackBoxModel id="OmegaRefPu" lib="Step" parFile="WT4ACurrentSource.par" parId="OmegaRefPu"/>
```
Comment (cossartque) [inline examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.dyd:19]: Same

File: examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.par [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, you can obtain one at http://mozilla.org/MPL/2.0/.
+    SPDX-License-Identifier: MPL-2.0
+
+    This file is part of Dynawo, an hybrid C++/Modelica open source suite of
+    simulation tools for power systems.
+-->
+<parametersSet xmlns="http://www.rte-france.com/dynawo">
+  <set id="WT4A">
+    <!-- Nominal -->
```
Comment (cossartque) [inline examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.par:16]: We should keep in mind to double check all the parameters with the full OM example

File: examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.par [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    <par type="DOUBLE" name="step_Height" value="-0.5"/>
+    <par type="DOUBLE" name="step_tStep" value="2"/>
+    <par type="DOUBLE" name="step_Value0" value="1"/>
+  </set>
+  <set id="QRefPu">
+    <par type="DOUBLE" name="step_Height" value="0.2"/>
+    <par type="DOUBLE" name="step_tStep" value="4"/>
+    <par type="DOUBLE" name="step_Value0" value="-0.21"/>
+  </set>
+  <set id="TanPhi">
```
Comment (cossartque) [inline examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.par:108]: useless

File: examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.par [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+    <par type="DOUBLE" name="step_Height" value="0"/>
+    <par type="DOUBLE" name="step_tStep" value="0"/>
+    <par type="DOUBLE" name="step_Value0" value="-0.21"/>
+  </set>
+  <set id="OmegaRefPu">
```
Comment (cossartque) [inline examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.par:113]: useless

**Review by cossartque — CHANGES_REQUESTED** [review]

File: examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.crv [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    This file is part of Dynawo, an hybrid C++/Modelica open source suite of
+    simulation tools for power systems.
+-->
+<curvesInput xmlns="http://www.rte-france.com/dynawo">
+  <curve model="WT4A" variable="WT_measurements_PFiltPu"/>
+  <curve model="WT4A" variable="WT_measurements_QFiltPu"/>
+  <curve model="WT4A" variable="WT_measurements_UPu"/>
+  <curve model="WT4A" variable="WT_wT4Ainjector_ipCmdPu"/>
+  <curve model="WT4A" variable="WT_wT4Ainjector_iqCmdPu"/>
+  <curve model="WT4A" variable="WT_wT4Ainjector_theta"/>
```
[the remark this answers is not in the document]
Comment (GerardThevenon) [reply examples/DynaSwing/IEC/Wind/Neplan/WT4ACurrentSource/WT4ACurrentSource.crv:20]: Unlike WECC, IEC PLL uses theta.

**Comment by cossartque** [comment]
Once this test case works, you should add the reference and the test case to the cases.py

**Review by cossartque — DISMISSED** [review]

File: nrt/data/DynaSwing/cases.py [hunk 6]
```diff
@@ -87,6 +87,16 @@
 
 test_cases.append((case_name, case_description, job_file, 5, standardReturnCodeType, standardReturnCode))
 
+###########################################
+#     IEC Wind Type 4A Current Source     #
```
Comment (GerardThevenon) [inline nrt/data/DynaSwing/cases.py:91]: Wind Turbine ?

File: nrt/data/DynaSwing/cases.py [hunk 7]
```diff
@@ -87,6 +87,16 @@
...
+###########################################
+
+case_name = "DynaSwing - IEC - Wind Type 4A - Current source"
+case_description = "IEC - Wind Type 4A - Current source"
```
Comment (GerardThevenon) [inline nrt/data/DynaSwing/cases.py:95]: Same

**Review by cossartque — APPROVED** [review]

**Pull request merged**